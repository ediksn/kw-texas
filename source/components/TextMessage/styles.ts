import { scale, verticalScale } from 'react-native-size-matters'
import { theme } from '~/constants'
import { TextColorsInterface } from '~/functions/textColors'

export const styles = ({ text, background }: TextColorsInterface) => ({
  container: {
    backgroundColor: background.light,
    borderColor: background.dark,
    paddingVertical: verticalScale(10),
    paddingHorizontal: scale(5),
    borderWidth: 1,
    borderRadius: 5
  },
  text: {
    color: text,
    fontSize: theme.fonts.MEDIUM_SIZE
  }
})
