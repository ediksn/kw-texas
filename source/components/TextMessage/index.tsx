import React from 'react'
import { Text, View, ViewStyle } from 'react-native'

import { styles } from './styles'
import { TextType } from '~/constants/theme'
import { textColors, TextColorsInterface } from '~/functions'

interface Props {
  message: string
  type: TextType
  style?: ViewStyle
}

const TextMessage = ({ message, type, style }: Props) => {
  const color: TextColorsInterface = textColors(type)
  return (
    <View style={[styles(color).container, style]}>
      <Text style={styles(color).text}>{message}</Text>
    </View>
  )
}

export default TextMessage
