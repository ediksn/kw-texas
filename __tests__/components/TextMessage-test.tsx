import React from 'react'
import renderer from 'react-test-renderer'
import TextMessage from '~/components/TextMessage'

test('renders correctly', () => {
  const header = renderer.create(<TextMessage type='error' message='This is a example message' />).toJSON()
  expect(header).toMatchSnapshot()
})
