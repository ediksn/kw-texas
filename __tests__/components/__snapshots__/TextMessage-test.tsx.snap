// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E9D1D2",
        "borderColor": "#E9BFC1",
        "borderRadius": 5,
        "borderWidth": 1,
        "paddingHorizontal": 10.714285714285714,
        "paddingVertical": 19.61764705882353,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#C5050B",
        "fontSize": 22,
      }
    }
  >
    This is a example message
  </Text>
</View>
`;
